pwndbg/aglib/kernel/vmmap.py

```diff
@@ -275,69 +345,35 @@ def kernel_vmmap_via_monitor_info_mem() -> Tuple[pwndbg.lib.memory.Page, ...]:
         monitor_info_mem = pwndbg.dbg.selected_inferior().send_monitor("info mem")
     except pwndbg.dbg_mod.Error:
         # Exception should not happen in new qemu, can we clean up it?
-        monitor_info_mem = None
-
-    is_error = monitor_info_mem is None or "unknown command" in monitor_info_mem
-    if is_error:
         # Older versions of QEMU/GDB may throw `gdb.error: "monitor" command
         # not supported by this target`. Newer versions will not throw, but will
-        # return a string starting with 'unknown command:'. We handle both of
-        # these cases in a `finally` block instead of an `except` block.
-        # TODO: Find out which other architectures don't support this command
-        if pwndbg.aglib.arch.name == "aarch64":
-            print(
-                M.error(
-                    f"The {pwndbg.aglib.arch.name} architecture does"
-                    " not support the `monitor info mem` command.\n"
-                    "Run `help show kernel-vmmap` for other options."
-                )
-            )
-        return ()
+        # return a string starting with 'unknown command:'.
+        monitor_info_mem = "unknown command"
 
-    lines = monitor_info_mem.splitlines()
+    parser_func = None
+    if pwndbg.aglib.arch.name in ("i386", "x86-64"):
+        parser_func = _parser_mem_info_line_x86
+    elif pwndbg.aglib.arch.name == "rv64":
+        parser_func = _parser_mem_info_line_riscv64
 
-    # Handle disabled PG
-    # This will prevent a crash on abstract architectures
-    if len(lines) == 1 and lines[0] == "PG disabled":
+    if parser_func is None or "unknown command" in monitor_info_mem:
+        print(
+            M.error(
+                f"The {pwndbg.aglib.arch.name} architecture does"
+                " not support the `monitor info mem` command.\n"
+                "Run `help show kernel-vmmap` for other options."
+            )
+        )
         return ()
 
-    global monitor_info_mem_not_warned
     pages: List[pwndbg.lib.memory.Page] = []
-    for line in lines:
+    for line in monitor_info_mem.splitlines():
         try:
-            dash_idx = line.index("-")
-            space_idx = line.index(" ")
-            rspace_idx = line.rindex(" ")
-
-            start = int(line[:dash_idx], 16)
-            end = int(line[dash_idx + 1 : space_idx], 16)
-            size = int(line[space_idx + 1 : rspace_idx], 16)
+            page = parser_func(line)
         except Exception:
             # invalid format
             continue
```

Comment:
We skip the exception if it happens, leaving user with no info that we may parsed sth wrong